src/common/src/util/epoch.rs

```diff
@@ -162,8 +162,131 @@ impl EpochPair {
     }
 
     pub fn new_test_epoch(curr: u64) -> Self {
-        assert!(curr > 0);
-        Self::new(curr, curr - 1)
+        assert!(curr > 65535);
+        Self::new(curr, curr - TestEpoch::new_without_offset(1).as_u64())
+    }
+}
+
+/// The `TestEpoch` struct is used in unit tests to provide consistent logic similar to a normal epoch.
+/// It ensures that the lower 16 bits are always zero and any addition or subtraction operations are only applied to the upper 48 bits.
+#[derive(Clone, Copy, PartialEq, Eq, Hash, Default, Debug, PartialOrd, Ord)]
+pub struct TestEpoch {
+    epoch_with_gap: TestEpochWithGap,
```

Comment:
We can, but in order to add more test for spill later, I'd like an independent struct similar to `EpochWithGap`, because `TestEpoch` need to provide some `inc`, `sub` method. But I don’t have a strong preference.